packages/compiler-core/src/parse.ts

```diff
@@ -768,15 +768,21 @@ function parseText(context: ParserContext, mode: TextModes): TextNode {
   __DEV__ && assert(context.source.length > 0)
 
   const [open] = context.options.delimiters
-  // TODO could probably use some perf optimization
-  const endIndex = Math.min(
-    ...[
-      context.source.indexOf('<', 1),
-      context.source.indexOf(open, 1),
-      mode === TextModes.CDATA ? context.source.indexOf(']]>') : -1,
-      context.source.length
-    ].filter(n => n !== -1)
-  )
+
+  const findBy = ['<', open]
+  if (mode === TextModes.CDATA) findBy.push(']]>')
+
+  let endIndex = Infinity
```

Comment:
I think we can start with `endIndex = context.source.length` since `indexOf` will never return a value greater than that

Reply:
Completely agree